hydra-node/test/Hydra/NodeSpec.hs

```diff
@@ -146,6 +146,23 @@ spec = parallel $ do
         runToCompletion tracer node
 
         getServerOutputs >>= (`shouldContain` [TxValid{headId = testHeadId, transaction = tx1}])
+  -- FIXME: this should be in a HeadLogicSpec
+  it "Emits ReqSn on valid RecDec" $
+    failAfter 1 $
+      showLogsOnFailure "NodeSpec" $ \tracer -> do
+        let events =
+              eventsToOpenHead
+                <> [ NetworkEvent
+                      { ttl = defaultTTL
+                      , party = bob
+                      , message = ReqDec{transaction = SimpleTx{txSimpleId = 1, txInputs = utxoRefs [2], txOutputs = utxoRefs [4]}}
+                      }
+                   ]
+
+        node <- createHydraNode bobSk [alice, carol] defaultContestationPeriod events
+        (node', getNetworkMessages) <- recordNetwork node
+        runToCompletion tracer node'
+        getNetworkMessages `shouldReturn` [ReqSn{snapshotNumber = 1, transactionIds = [1], decommitTx = Nothing}]
```

Comment:
i do not understand, why if it valid the ReqDec we have a Snapshot with Nothing in decommitTx?

Reply:
Good point. This test was actually red since the logic behind it is already there ad the decommitTx is populated. One thing I left a TODO for is why `transactionIds` is empty list instead of containing the tx id.

Reply:
And in any case `transactionIds` should stay empty since we don't record what we put on L1.